### PR TITLE
Python: Add module boundary flow steps

### PR DESCRIPTION
**Pull request by tausbn**
This is the quick-and-dirty solution, as discussed.

An even quicker-and-dirtier solution would have used
`ModuleValue::attr` and take the `getOrigin` of that as the source of
the jump step. However, this turns out to be a bad choice, since
`attr` might fail to have a value for the given attribute (for a
variety of reasons). Thus, we instead appeal to a helper predicate
that keeps track of which names are defined by which right-hand-sides
in a given module. (Observe that type tracking works correctly for `x`
in `mymodule.py`, even though `x` is never assigned a value in the
eyes of the Value API.)

This means that points-to is only used to actually figure out if the
object we're looking an attribute up on is a module or not. This is
the next thing to replace in order to eliminate the dependence on
points-to, but this will require some care to ensure that all module
lookups are handled correctly.

Only two test files needed to be changed for the tests to pass. The
first was the fixed false negative in the type tracker, and the other
was a bunch of missing flow in the regression test. I have manually
removed the `# Flow not found` annotations to make them consistent
with the output. Pay particular attention to the annotation on line
117 -- I believe it was misplaced and should have been on line 106
instead (where, indeed, we now have flow where none appeared before).

Finally, this PR adds a bunch of test files based on PEP-328 in anticipation 
of the import resolution handling that will be needed eventually. The actual
test will come at a later point. (Probably not this PR.)

### CODE REVIEW

File: python/ql/src/experimental/dataflow/internal/DataFlowPrivate.qll [hunk 1]
```diff
@@ -504,6 +504,26 @@ predicate jumpStep(Node nodeFrom, Node nodeTo) {
...
+  // Read of module attribute:
+  exists(AttrRead r, ModuleValue mv |
+    r.getObject().asCfgNode().pointsTo(mv) and
+    module_export(mv.getScope(), r.getAttributeName(), nodeFrom) and
+    nodeTo = r
+  )
+}
+
+/**
+ * Holds if the module `m` defines a name `name` that by assigning `defn` to it. This is an
```
Comment (yoff) [inline python/ql/src/experimental/dataflow/internal/DataFlowPrivate.qll:517]:
```suggestion
 * Holds if the module `m` defines a name `name` by assigning `defn` to it. This is an
```

**Review by yoff — APPROVED** [review]
One simple typo, otherwise looks good. It does look like you are correct about the test annotations. It will be glorious once all test annotations are verified :-)

File: python/ql/src/experimental/dataflow/internal/DataFlowPrivate.qll (hunk 1 above)
Comment (tausbn) [reply python/ql/src/experimental/dataflow/internal/DataFlowPrivate.qll]: Good catch!

**Review by yoff — APPROVED** [review]
LGTM 👍

**Review by RasmusWL — APPROVED** [review]
LGTM!

I'll just run this on a snapshot locally, and if everything looks good, merge it as well :+1:

**Review by RasmusWL — CHANGES_REQUESTED** [review]
Running it against real snapshot showed that we did not quite handle my case. I figured out the problem, and added testcase in https://github.com/tausbn/codeql/pull/2

**Review by RasmusWL — APPROVED** [review]
Nice :muscle: works on the real project I mentioned in https://github.com/tausbn/codeql/pull/2 as well :tada:

**Review by yoff — APPROVED** [review]
Interesting, it seems we might do this in a more principled way later, by referring to `ModuleExport`s, perhaps. But this is fine as bandaid for now.

**Pull request merged**